test-data/src/main/java/dev/futa/tutorial/pesel/PeselDataSetSupplier.java

```diff
@@ -0,0 +1,41 @@
+package dev.futa.tutorial.pesel;
+
+import com.google.common.base.Preconditions;
+
+import java.time.LocalDate;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static dev.futa.tutorial.pesel.Gender.FEMALE;
+import static dev.futa.tutorial.pesel.Gender.MALE;
+
+public enum PeselDataSetSupplier {
+  INSTANCE;
```

Comment:
Nie rozumiem dlaczego `enum,` a nie zwykła klasa

Reply:
Nie ma to szczególnego sensu poza uzyskaniem... singletona, na którego nie ma potrzeby.

Chciałem uniknąć później przekazania i tworzenia klasy za pomocą _new_. To też mi się nie podobało. 

Ale masz racę, że w @BeforeAll to trzeba ułatwić.

Reply:
`PeselDataSetSupplier.INSTANCE.getValidPeselSet()` vs `PeselDataSetSupplier.getValidPeselSet()`. Można zrobić statyczną metodę `getValidPeselSet()`, albo nawet stałą `ImmutableSet`.